src/ks_log.c

```diff
@@ -307,19 +260,26 @@ static void default_logger(const char *file, const char *func, int line, int lev
 	
 	va_start(ap, fmt);
 
-	// Prefix the buffer with the color
-	strcpy(buf, LEVEL_COLORS[level]);
-	prefixlen = strlen(buf);
-	suffixlen = strlen(KNRM);
-
-	// Add to the buffer after the color, save space for color reset
-	len = ks_log_format_output(buf + prefixlen, sizeof(buf) - prefixlen - suffixlen, file, func, line, level, fmt, ap);
+	len = ks_log_format_output(buf, sizeof(buf), file, func, line, level, fmt, ap);
 
 	if (len > 0) {
+		if (ks_log_jsonified) {
+			ks_json_t *json = ks_json_create_object();
+			ks_json_add_string_to_object(json, "message", buf);
+			char *tmp = ks_json_print_unformatted(json);
+			strncpy(buf, tmp, sizeof(buf) - 2); // safe truncation if neccessary, leaving 2 bytes for newline and terminator
```

Comment:
You could just fwrite tmp then fwrite ('\n') instead.  One less copy and no chance of truncating the JSON log.

Reply:
fprintf() probably an even better choice for JSON output

Reply:
I was trying to avoid restructuring all the logging code, but we can go that way.